### PR TITLE
add GCloud install details and move to new file

### PR DESCRIPTION
**Pull request by misaka**
Add a bunch more detail to the instructions for setting up GCloud and move these instructions to a separate file.

This is a fairly messy change, soz. The key changes are:

- move the instructions for setting up Google Cloud into a separate file `docs/GOOGLE_CLOUD_BIGQUERY_SETUP.md`
- change Gemfile example to use GitHub
- recommend creating a separate service account for each dataset instead of a single service account with access to all the datasets, plus additional explanations and examples
- add more words to make it clearer what each step is doing
- refer to Google Cloud over just "BigQuery". e.g. the project created is a Google Cloud project, not just a BigQuery project. This means these projects will have access to Google Maps APIs and other services in the GCP suite.

### CODE REVIEW

**Review by ethax-ross — COMMENTED** [review]
These changes look great and it all makes sense; just a few minor things I noticed on reading through it.

File: README.md [hunk 1]
```diff
@@ -344,7 +111,7 @@ BIGQUERY_DATASET=your-bigquery-dataset-name
 BIGQUERY_API_JSON_KEY=<contents of the JSON, make sure to strip or escape newlines>
```
[the remark this answers is not in the document]
Comment (misaka) [reply README.md:111]: Let's have a look at this, we've gotten it to work by adjusting the quoting in the past.

File: README.md [hunk 2]
```diff
@@ -74,265 +86,20 @@ then
...
-### 4. Create an appender service account
-
-1. Go to [IAM and Admin settings > Create service account](https://console.cloud.google.com/projectselector/iam-admin/serviceaccounts/create?supportedpurview=project)
-1. Name it like "Appender NAME_OF_SERVICE ENVIRONMENT" e.g. "Appender ApplyForQTS Production"
-1. Add a description, like "Used when developing locally."
-1. Grant the service account access to the project, use the "BigQuery Appender Custom" role you set up earlier
+### 2. Get an API JSON key :key:
 
-### 5. Get an API JSON key :key:
+Depending on how your app environments are setup, we recommend you use the
```
[the remark this answers is not in the document]
Comment (misaka) [reply README.md:91]: I tried adding something simple to explain this.

**Review by ethax-ross — APPROVED** [review]
👏

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (misaka) [reply README.md:113]: As discussed, this was due to `YAML.dump` inserting wrapping long lines as used by `fetch_config.rb`❗🤦‍♂️ . Hopefully this is something we can address in `fetch_config.rb`, but for now Base64 or just not using `fetch_config.rb` for local dev envs are workarounds.

File: README.md [hunk 3]
```diff
@@ -64,8 +65,19 @@ A Rails app with `ActiveJob` configured.
 
 ## Installation
 
+Before you can send data to BigQuery with `dfe-analytics` you'll need to setup
+your Google Cloud project. See the [setup Google Cloud setup guide](docs/GOOGLE_CLOUD_BIGQUERY_SETUP.md)
```
Comment (thomasleese) [inline README.md:69]:
```suggestion
your Google Cloud project. See the [setup Google Cloud setup guide](docs/google-cloud-bigquery-setup.md)
```

Feels a little odd to have the filename all in caps, but not a big deal.

**Review by thomasleese — APPROVED** [review]
Looks good, I've only got one comment, but it's not a big thing!

File: README.md (hunk 3 above)
Comment (misaka) [reply README.md:69]: Ya, agreed, will change.

**Comment by misaka** [comment]
[Updates README for dfe-analytics.](https://trello.com/c/AbOFhspb/795-updates-readme-for-dfe-analytics)

**Pull request merged**